cloud/openstack/os_nova_flavor.py

```diff
@@ -206,6 +216,12 @@ def main():
 
         if state == 'present':
             if not flavor:
+                if type(module.params['extra_specs']) is str:
+                    extra_specs = {}
+                    for kv_str in module.params['extra_specs'].split(","):
+                        k, v = kv_str.split("=")
+                        extra_specs[k] = v
+                    module.params['extra_specs'] = extra_specs
```

Comment:
From reading what the Ansible module helpers do with a type of dict, this bit should be done for you. Have you tested this out without putting in the string conversion code?